lib/util.js

```diff
@@ -1057,3 +1057,72 @@ exports._exceptionWithHostPort = function(err,
 // process.versions needs a custom function as some values are lazy-evaluated.
 process.versions[exports.inspect.custom] =
   (depth) => exports.format(JSON.parse(JSON.stringify(process.versions)));
+
+
+/**
+ * @typedef {Object} callbackifyOptions
+ * @property {function} transformer
+ * @property {boolean} toSplat
+ * @property {Array.<String>} project
+ */
+
+/**
+ * @param {function(): Promise} orig
+ * @param {callbackifyOptions} options
+ */
+function callbackify(orig, options) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to callbackify() ' +
+                        'must be a function');
+  }
+
+  // we have two way to convert the returned value into multiple arguments
+  // either splat it or project some of it's keys
+  const toSplat = options.toSplat;
```

Comment:
I don't think we should support this - with `promisify` we don't really have a choice for things like `fs.exists` but here destructuring syntax should be sufficient.